### PR TITLE
Document DD_EXTRA_TAGS

### PR DESCRIPTION
**Pull request by djmitche**
This should (hopefully!) make the meaning of this config value clear.

### Motivation

More docs

### Additional Notes

Originally landed in #7872.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -106,14 +106,30 @@ api_key:
 ## @env DD_TAGS - space separated list of strings - optional
 ## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
 ##
-## Additional tags can be supplied using the `DD_EXTRA_TAGS` environment variable.
+## This configuration value is merged with DD_EXTRA_TAGS.  This allows some
+## tags to be set in a configuration file (tags), with additional tags added
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml:110]: Trying to make it clearer that this instance of `tags` refers to the config option name:

```suggestion
## tags to be set in a configuration file (`tags`), with additional tags added
```

File: pkg/config/config_template.yaml [hunk 2]
```diff
@@ -106,14 +106,30 @@ api_key:
...
 # tags:
 #   - team:infra
 #   - <TAG_KEY>:<TAG_VALUE>
 
+## @param extra_tags  - list of key:value elements - optional
+## @env DD_EXTRA_TAGS - space separated list of strings - optional
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+##
+## This configuration value is merged with DD_TAGS.  This allows some
+## tags to be set in a configuration file (tags), with additional tags added
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml:124]:
```suggestion
## tags to be set in a configuration file (`tags`), with additional tags added
```

File: pkg/config/config_template.yaml [hunk 3]
```diff
@@ -106,14 +106,30 @@ api_key:
...
 #
 # tags:
 #   - team:infra
 #   - <TAG_KEY>:<TAG_VALUE>
 
+## @param extra_tags  - list of key:value elements - optional
+## @env DD_EXTRA_TAGS - space separated list of strings - optional
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+##
+## This configuration value is merged with DD_TAGS.  This allows some
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml:123]:
```suggestion
## This configuration value is merged with `tags`.  This allows some
```

**Review by olivielpeau — APPROVED** [review]
thanks for clarifying this 👍

File: pkg/config/config_template.yaml [hunk 4]
```diff
@@ -106,14 +106,30 @@ api_key:
 ## @env DD_TAGS - space separated list of strings - optional
 ## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
 ##
-## Additional tags can be supplied using the `DD_EXTRA_TAGS` environment variable.
+## This configuration value is merged with DD_EXTRA_TAGS.  This allows some
+## tags to be set in a configuration file (`tags`), with additional tags added
+## via an environment variable (DD_EXTRA_TAGS).
```
Comment (alai97) [inline pkg/config/config_template.yaml:111]:
```suggestion
## This configuration value merges with `DD_EXTRA_TAGS`, allowing some
## tags to be set in a configuration file (`tags`), and additional tags to be added
## with an environment variable (`DD_EXTRA_TAGS`).
```

File: pkg/config/config_template.yaml [hunk 5]
```diff
@@ -106,14 +106,30 @@ api_key:
...
 #   - team:infra
 #   - <TAG_KEY>:<TAG_VALUE>
 
+## @param extra_tags  - list of key:value elements - optional
+## @env DD_EXTRA_TAGS - space separated list of strings - optional
+## List of host tags. Attached in-app to every metric, event, log, trace, and service check emitted by this Agent.
+##
+## This configuration value is merged with `tags`.  This allows some
+## tags to be set in a configuration file (`tags`), with additional tags added
+## via an environment variable (DD_EXTRA_TAGS).
```
Comment (alai97) [inline pkg/config/config_template.yaml:125]:
```suggestion
## This configuration value merges with `tags`, allowing some
## tags to be set in a configuration file (`tags`), and additional tags to be added
## with an environment variable (`DD_EXTRA_TAGS`).
```

**Review by alai97 — APPROVED** [review]
Looks good, two copy nits!

**Pull request merged**